app/controllers/users_controller.rb

```diff
@@ -251,6 +249,8 @@ def account_report
   end
 
   def reset_password
+    @user = User.find(params[:id])
+    authorize @user, :update_edit_or_destroy?
```

Comment:
Done.  Added a UserPolicy@reset_password?:

      def reset_password?
        changeable?
      end

And changed controller to:

      def reset_password
        @user = User.find(params[:id])
        authorize @user